dialogs/Options.cpp

```diff
@@ -1,73 +1,868 @@
+
+	m_pPrefabDir = new CFileSelector( this, CFileSelector::DirsOnly );
+	m_pPrefabDir->getLineEdit().setText( "path/to/prefabs" );
+	m_pPrefabDir->getLabel().setText( tr( "Prefabs Directory (ex: c:\\Half-Life\\maps\\prefabs):" ) );
+
+	pWidgetLayout->addLayout( addConfigAndGameDataLayout() );
+	pWidgetLayout->addLayout( addEntityAndScaleLayout() );
+	pWidgetLayout->addWidget( m_pCordonTexture );
+	pWidgetLayout->addWidget( m_pGameExec );
+	pWidgetLayout->addWidget( m_pGameDir );
+	pWidgetLayout->addWidget( m_pHammerVmfDir );
+	pWidgetLayout->addWidget( m_pPrefabDir );
+
+	this->setLayout( pWidgetLayout );
+
+	this->setFixedSize( this->sizeHint() );
+}
+
+QGridLayout *COptionsGameConfigsTab::addConfigAndGameDataLayout()	
+{
+	auto pConfigLayout = new QGridLayout();
+
+	auto pConfigLabel = new QLabel( tr( "Configuration:" ) );
+	m_pConfigCombo = new QComboBox( this );
+	m_pConfigCombo->addItem( "Momentum" );
+	auto pConfigEditButton = new QPushButton( tr( "Edit" ), this );
+
+	auto pGameDataLabel = new QLabel( tr( "Game Data files:" ), this );
+	m_pGameDataList = new QListWidget( this );
+	m_pGameDataList->setFixedHeight( 50 );
+	m_pGameDataList->addItem( "path/to/momentum.fgd" );
+
+	auto pButtonLayout = new QVBoxLayout();
+	auto pAddEditLayout = new QHBoxLayout();
+	auto pGameDataAddButton = new QPushButton( tr( "Add" ), this );
+	auto pGameDataEditButton = new QPushButton( tr( "Edit" ), this );
+	auto pRemoveButton = new QPushButton( tr( "Remove" ), this );
 
-	auto pGameExeSelect = new CFileSelector( this, CFileSelector::DirsOnly );
-	pGameExeSelect->getLabel().setText( "Game Executable Directory:" );
+	pAddEditLayout->addWidget( pGameDataAddButton );
+	pAddEditLayout->addWidget( pGameDataEditButton );
 
-	auto pGameDirSelect = new CFileSelector( this, CFileSelector::DirsOnly );
-	pGameDirSelect->getLabel().setText( "Game Directory (where GameInfo.txt is):" );
+	pGameDataAddButton->setFixedWidth( pRemoveButton->width() / 2 );
+	pGameDataEditButton->setFixedWidth( pRemoveButton->width() / 2 );
+
+	pButtonLayout->addLayout( pAddEditLayout );
+	pButtonLayout->addWidget( pRemoveButton );
+	
+	// Config
+	pConfigLayout->addWidget( pConfigLabel, 0, 0 );
+	pConfigLayout->addWidget( m_pConfigCombo, 1, 0, 1, 3 );
+	pConfigLayout->addWidget( pConfigEditButton, 1, 4 );
+
+	// Game Data
+	pConfigLayout->addWidget( pGameDataLabel, 2, 0 );
+	pConfigLayout->addWidget( m_pGameDataList, 3, 0, 1, 3 );
+	pConfigLayout->addLayout( pButtonLayout, 3, 4, 1, 1, Qt::AlignTop );
+
+	connect( pConfigEditButton, &QPushButton::released, this, &COptionsGameConfigsTab::onConfigEditPressed );
+	connect( pGameDataAddButton, &QPushButton::released, this, &COptionsGameConfigsTab::onGameDataAddPressed );
+	connect( pGameDataEditButton, &QPushButton::released, this, &COptionsGameConfigsTab::onGameDataEditPressed );
+	connect( pRemoveButton, &QPushButton::released, this, &COptionsGameConfigsTab::onGameDataRemovePressed );
```

Comment:
Is there any particular reason for connecting to `QPushButton::released` instead of `QPushButton::clicked`?

Reply:
Good question, don't think there is!

Reply:
It was a while ago when I did this, but I think I just copied the same way that pretty much all the other dialogs did it, which was just using `released` instead of `clicked`. If it's an issue, I can easily change it though

Reply:
It can stay as is for this PR, and change `released` to `clicked` events for all files in separate PR.

Reply:
Released mimicks the behavior of most buttons, that's why I chose it. (ex: click and hold a push button in github's UI, move the cursor off and release, the button's event won't be fired. with `clicked` the button's event fires immediately)

Reply:
On windows it `clicked` fires after I release the button.